lib/src/slices/data_bkg/data_bkg_service.dart

```diff
@@ -4,274 +4,83 @@
-              model.gmailCategoryList[0];
-      int categoryStartIndex = model.gmailCategoryList.indexOf(activeCategory);
-      int totalCats = model.gmailCategoryList.length;
-      for (int i = categoryStartIndex; i < totalCats; i++) {
-        String category = model.gmailCategoryList[i];
-        String query = _gmailBuildQuery(gmailLastFetchEpoch, category);
-        await _checkGmailFetchList(query: query);
-        int nextCatIndex = model.gmailCategoryList.indexOf(category) + 1;
-        String nextCat = nextCatIndex == totalCats
-            ? model.gmailCategoryList[0]
-            : model.gmailCategoryList[nextCatIndex];
-        await _apiAppDataService.save(ApiAppDataKey.gmailCategory, nextCat);
-      }
+  late final DataBkgServiceEmail email;
+
+  DataBkgService(
+      {required ApiAuthService apiAuthService,
+      required ApiAppDataService apiAppDataService,
+      required ApiCompanyService apiCompanyService,
+      required ApiEmailSenderService apiEmailSenderService,
+      required ApiEmailMsgService apiEmailMsgService})
+      : this._apiAppDataService = apiAppDataService,
+        this._apiAuthService = apiAuthService,
+        this.email = DataBkgServiceEmail(
+            apiAuthService: apiAuthService,
+            apiAppDataService: apiAppDataService,
+            apiCompanyService: apiCompanyService,
+            apiEmailSenderService: apiEmailSenderService,
+            apiEmailMsgService: apiEmailMsgService);
+
+  Future<void> index(List<ApiAuthServiceAccountModel> accounts) async {
+    _log.fine("fetch all data started");
+    int startIndex = await _getStartIndex();
+    for (int i = startIndex; i < accounts.length; i++) {
+      ApiAuthServiceAccountModel account = accounts[i];
+      await fetchData(account);
       await _apiAppDataService.save(
-          ApiAppDataKey.gmailLastFetch, run.millisecondsSinceEpoch.toString());
-      await _apiAppDataService.save(ApiAppDataKey.gmailPage, '');
-      _log.fine(
-          'Gmail fetch completed on: ' + DateTime.now().toIso8601String());
-    }
-  }
-
-  Future<void> _checkGmailFetchList({String query = ''}) async {
-    String? page;
-    ApiAppDataModel? appDataGmailPage =
-        await _apiAppDataService.getByKey(ApiAppDataKey.gmailPage);
-    page = appDataGmailPage?.value;
-    return _checkGmailFetchListPage(query: query, page: page);
-  }
-
-  Future<void> _checkGmailFetchListPage({String? page, String? query}) async {
-    DataBkgModelPage<String> res = await _apiGoogleService.gmailFetch(
-        query: query, page: page, maxResults: 5);
-    if (res.data != null) {
-      List<String> known =
-          (await _apiEmailMsgService.getByExtMessageIds(res.data!))
-              .map((message) => message.extMessageId!)
-              .toList();
-      List<String> unknown =
-          res.data!.where((message) => !known.contains(message)).toList();
-      await _checkGmailFetchMessage(unknown);
+          ApiAppDataKey.dataBkgLastAccount, i.toString());
```

Comment:
We should not be storing last account by position in a list. There's no guarantee of order. You could store the the unique id of the account and then sort the list alphabetically or you could serialize a comma separated list of accounts that still need to be checked after each run.